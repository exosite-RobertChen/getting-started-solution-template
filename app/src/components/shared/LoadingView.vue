<template>
  <v-row justify="center" align="center">
    <v-col class="text-center">
      <v-progress-circular :size="50" color="primary" indeterminate />
      <div class="mt-6" :class="textColorClass">{{ text }}</div>
    </v-col>
  </v-row>
</template>

<script>
export default {
  props: {
    text: {
      type: String,
      required: false,
      default: ''
    },
    textColorClass: {
      type: String,
      required: false,
      default: ''
    }
  }
}
</script>
