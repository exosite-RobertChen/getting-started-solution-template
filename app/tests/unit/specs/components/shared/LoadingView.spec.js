import { shallowComponent } from '@/../tests/unit/common'
import LoadingView from '@/components/shared/LoadingView.vue'

const factory = overrides => shallowComponent(LoadingView, overrides)

describe('LoadingView.vue', () => {
  test('can be mounted', () => {
    const wrapper = factory()
    expect(wrapper.contains(LoadingView)).toBe(true)
  })

  test('matches snapshot', () => {
    const wrapper = factory()

    expect(wrapper.element).toMatchSnapshot()
  })
})
