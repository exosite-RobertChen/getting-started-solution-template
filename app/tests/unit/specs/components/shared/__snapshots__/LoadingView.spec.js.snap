// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingView.vue matches snapshot 1`] = `
<v-row-stub
  align="center"
  justify="center"
  tag="div"
>
  <v-col-stub
    class="text-center"
    tag="div"
  >
    <v-progress-circular-stub
      color="primary"
      indeterminate="true"
      rotate="0"
      size="50"
      value="0"
      width="4"
    />
     
    <div
      class="mt-6"
    >
      
    </div>
  </v-col-stub>
</v-row-stub>
`;
